teuthology/task/kernel.py

```diff
@@ -354,31 +353,19 @@ def download_kernel(ctx, config):
             procs[role_remote.name] = proc
         else:
             # gitbuilder package - src is sha1
-            log.info('Downloading kernel {sha1} on {role}...'.format(sha1=src,
-                                                                     role=role))
+            log.info('Downloading kernel {sha1} on {role}...'.format(
+                sha1=src,
+                role=role,
+            ))
             needs_download = True
-            package_type = role_remote.os.package_type
-            if package_type == 'rpm':
-                system_type = role_remote.os.name
-                system_ver = role_remote.os.version
-                if '.' in system_ver:
-                   system_ver = system_ver.split('.')[0]
-                ldist = '{system_type}{system_ver}'.format(
-                    system_type=system_type, system_ver=system_ver)
-                larch = 'x86_64'
-            elif package_type == 'deb':
-                ldist, larch = role_remote.os.codename, role_remote.arch
-            else:
-                raise UnsupportedPackageTypeError(role_remote)
 
-            _, baseurl = teuthology.get_ceph_binary_url(
-                package='kernel',
-                sha1=src,
-                format=package_type,
-                flavor='basic',
-                arch=larch,
-                dist=ldist,
-                )
+            gitbuilder = GitbuilderProject(
+                'kernel',
+                ctx.config.get('kernel'),
+                ctx=ctx,
+                remote=role_remote,
+            )
+            baseurl = gitbuilder.base_url + "/"
```

Comment:
So, does that mean it should be safe to supply the trailing ``/``?

On second thought, I don't want to block this bugfix for this. We should look into this at some point though.